tests/src/test/java/org/eclipse/hono/tests/amqp/AmqpConnectionIT.java

```diff
@@ -87,6 +87,26 @@ public void testConnectSucceedsForRegisteredDevice(final VertxTestContext ctx) {
         .setHandler(ctx.completing());
     }
 
+    /**
+     * Verifies that the device gets created when auto-provisioning is enabled.
+     *
+     * @param ctx The test context.
+     */
+    @Test
+    public void testAutoProvisioning(final VertxTestContext ctx) {
+        final String tenantId = helper.getRandomTenantId();
+        final SelfSignedCertificate deviceCert = SelfSignedCertificate.create(UUID.randomUUID().toString());
+
+        helper.getCertificate(deviceCert.certificatePath())
+                .compose(cert -> {
+                    final var tenant = Tenants.createTenantForTrustAnchor(cert);
+                    tenant.getTrustedCertificateAuthorities().get(0).setAutoProvisioningEnabled(true);
+                    return helper.registry.addTenant(tenantId, tenant);
+                })
+                .compose(ok -> connectToAdapter(deviceCert))
+                .setHandler(ctx.completing());
```

Comment:
there seem to be assertions missing here:
* device identity has been created
* credentials have been created

Reply:
Wouldn't that belong to an integration test of the device registry? Here I would rather add a test that the connect *does not* succeed if auto-provisioning is disabled for the trust nachor.

Reply:
That  would make sense. Do you have a corresponding IT for the registry?

Reply:
I'm working on it